crates/core/tedge_mapper/src/main.rs

```diff
@@ -101,9 +101,14 @@ async fn main() -> anyhow::Result<()> {
     };
     set_log_level(log_level);
 
-    // Run only one instance of a mapper
-    let run_dir: PathBuf = config.query(RunPathSetting)?.into();
-    let _flock = check_another_instance_is_not_running(&mapper_opt.name.to_string(), &run_dir)?;
+    // Run only one instance of a mapper (if enabled)
+    if config.query(LockFilesSetting)?.is_set() {
+        let run_dir: PathBuf = config.query(RunPathSetting)?.into();
+        let _flock = Some(check_another_instance_is_not_running(
+            &mapper_opt.name.to_string(),
+            &run_dir,
+        )?);
+    }
```

Comment:
The lock must stay in scope up to the end of `main()`, as it would be released on drop.

```suggestion
    let mut flock = None;
    if config.query(LockFilesSetting)?.is_set() {
        let run_dir: PathBuf = config.query(RunPathSetting)?.into();
        flock = Some(check_another_instance_is_not_running(
            &mapper_opt.name.to_string(),
            &run_dir,
        )?);
    }
```

Reply:
Edited